test/boost/statement_restrictions_test.cc

```diff
@@ -359,3 +359,136 @@ SEASTAR_TEST_CASE(slice_single_column_mixed_order) {
                     singular({I(1), I(1)}), singular({I(1), I(2)})}));
     });
 }
+
+// Currently expression doesn't have operator==().
+// Implementing it is ugly, because there are shared pointers and the term base class.
+// For testing purposes checking stringified expressions is enough.
+static bool expression_eq(const expr::expression& e1, const expr::expression& e2) {
+    return to_string(e1) == to_string(e2);
+}
+
+static void assert_expr_vec_eq(const std::vector<expr::expression>& v1, const std::vector<expr::expression>& v2) {
+    if (std::equal(v1.begin(), v1.end(), v2.begin(), v2.end(), expression_eq)) {
+        return;
+    }
+
+    BOOST_FAIL(fmt::format("Expression vectors not equal! [{}] != [{}]", fmt::join(v1, ", "), fmt::join(v2, ", ")));
+}
+
+// Unit tests for extract_column_restrictions function
+BOOST_AUTO_TEST_CASE(expression_extract_column_restrictions) {
+    using namespace expr;
+
+    auto make_column = [](const char* name, column_kind kind, int id) -> column_definition {
+        column_definition definition(name, int32_type, kind, id);
+
+        ::shared_ptr<column_identifier> identifier = ::make_shared<column_identifier>(name, true);
+        column_specification specification("ks", "cf", std::move(identifier), int32_type);
```

Comment:
`column_definition` has to have `column_specifiction` because `to_string` uses it for column name

Reply:
Why not explain that with a comment, then? ;)